lib/apis/moray.js

```diff
@@ -631,35 +660,121 @@ Moray.prototype._addFilterOptions = function (params) {
         }
 
         if (splitted.length == 2) {
-            order = splitted[1] || defaultOrder;
+            order = splitted[1] || common.DEFAULT_SORT_ORDER;
         }
 
-        options.sort = {
+        sortOptions.push({
             attribute: property,
             order: order
-        };
-    } else {
-        options.sort = {
-            attribute: '_id',
-            order: 'ASC'
-        };
+        });
     }
 
-    if (params.offset) {
-        options.offset = params.offset;
+    // If the sorting options will not output a totally ordered results set,
+    // add a sort option that will. This makes the last item of any
+    // results set (even when not using a marker, for example when getting
+    // the first page of paginated results) usable as a marker for getting
+    // the next page of paginated results.
+    var sortOptionsHasStrictTotalOrder =
+        sortOptions.some(function (sortOption) {
+            return common.isStrictTotalOrderField(sortOption.attribute);
+        });
+
+    if (!sortOptionsHasStrictTotalOrder) {
+        sortOptions.push({
+            attribute: common.strictTotalOrderField(),
+            order: common.DEFAULT_SORT_ORDER
+        });
     }
 
-    if (params.limit) {
-        options.limit = params.limit;
+    return sortOptions;
+}
+
+/*
+ * Build the appropriate LDAP filters for the marker "marker"
+ * and the sort options "sortOptions". Returns a list of strings
+ * representing ldap filters. An empty list is returned if not filters
+ * needs to be used for the corresponding marker and sortOptions.
```

Comment:
Fixed in [the latest commit of this PR's branch](https://github.com/misterdjules/sdc-vmapi/commit/5d65624f5bf76038e74fd8c831bcf56a6f42102f#diff-39ed83153bb1256b9859e9b2d507c2b0R699).